core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2014,8 +2014,10 @@ private[spark] object Utils extends Logging {
       } catch {
         case e: Exception if isBindCollision(e) =>
           if (offset >= maxRetries) {
-            val exceptionMessage =
-              s"${e.getMessage}: Service$serviceString failed after $maxRetries retries!"
+            val exceptionMessage = s"${e.getMessage}: Service$serviceString failed after " +
+              s"$maxRetries retries! Consider explicitly setting the appropriate port for" +
```

Comment:
Man, that's tricky.

But in the case where `serviceString` is empty, this would look odd:

    Consider explicitly setting the appropriate port for (for example spark.ui.port for SparkUI)

Perhaps `serviceString` should be changed so that it's "Service" if `serviceName` is empty. Or make sure that it's never empty.